controlplane/kubeadm/internal/controllers/status_test.go

```diff
@@ -978,11 +1200,41 @@ func Test_setAvailableCondition(t *testing.T) {
 					},
 				},
 				Machines: collections.FromMachines(
-					&clusterv1.Machine{ObjectMeta: metav1.ObjectMeta{Name: "m1"}, Status: clusterv1.MachineStatus{V1Beta2: &clusterv1.MachineV1Beta2Status{Conditions: []metav1.Condition{apiServerPodNotHealthy, controllerManagerPodHealthy, schedulerPodHealthy, etcdPodHealthy, etcdMemberHealthy}}}},
-					&clusterv1.Machine{ObjectMeta: metav1.ObjectMeta{Name: "m2"}, Status: clusterv1.MachineStatus{V1Beta2: &clusterv1.MachineV1Beta2Status{Conditions: []metav1.Condition{apiServerPodHealthy, controllerManagerPodHealthy, schedulerPodHealthy, etcdPodHealthy, etcdMemberHealthy}}}},
-					&clusterv1.Machine{ObjectMeta: metav1.ObjectMeta{Name: "m3"}, Status: clusterv1.MachineStatus{V1Beta2: &clusterv1.MachineV1Beta2Status{Conditions: []metav1.Condition{apiServerPodNotHealthy, controllerManagerPodHealthy, schedulerPodHealthy, etcdPodHealthy, etcdMemberHealthy}}}},
+					&clusterv1.Machine{
+						ObjectMeta: metav1.ObjectMeta{Name: "m1"},
+						Status: clusterv1.MachineStatus{
+							NodeRef: &corev1.ObjectReference{Name: "m1"},
+							V1Beta2: &clusterv1.MachineV1Beta2Status{Conditions: []metav1.Condition{apiServerPodHealthy, controllerManagerPodHealthy, schedulerPodHealthy, etcdPodHealthy, etcdMemberHealthy}},
+						},
+					},
+					&clusterv1.Machine{
+						ObjectMeta: metav1.ObjectMeta{Name: "m2"},
+						Status: clusterv1.MachineStatus{
+							NodeRef: &corev1.ObjectReference{Name: "m2"},
+							V1Beta2: &clusterv1.MachineV1Beta2Status{Conditions: []metav1.Condition{apiServerPodHealthy, controllerManagerPodHealthy, schedulerPodHealthy, etcdPodHealthy, etcdMemberHealthy}},
+						},
+					},
+					&clusterv1.Machine{
+						ObjectMeta: metav1.ObjectMeta{Name: "m3"},
+						Status: clusterv1.MachineStatus{
+							NodeRef: &corev1.ObjectReference{Name: "m3"},
+							V1Beta2: &clusterv1.MachineV1Beta2Status{Conditions: []metav1.Condition{apiServerPodHealthy, controllerManagerPodHealthy, schedulerPodHealthy, etcdPodHealthy, etcdMemberNotHealthy11s}},
+						},
+					},
+					&clusterv1.Machine{
+						ObjectMeta: metav1.ObjectMeta{Name: "m4"},
+						Status: clusterv1.MachineStatus{
+							NodeRef: nil,
+							V1Beta2: &clusterv1.MachineV1Beta2Status{Conditions: []metav1.Condition{apiServerPodHealthy, controllerManagerPodHealthy, schedulerPodHealthy, etcdPodHealthy, etcdMemberHealthy}},
```

Comment:
added a note with also the explanation why adding a use case with unknown conditions does not provide a good signal

Reply:
All good for this test case.

Would it be good to have an additional test case with the realistic case of a Machine without nodeRef and unknown conditions? (because this is something that actually happens, so it would be good to cover)